spec.bs

```diff
@@ -3418,47 +3445,57 @@ algorithms to achieve the outcomes described in the above explanatory content.
 
-  Given null or an [=element=] (|container|), this algorithm returns a new [=permissions policy=].
+  Given an [=element=] (|container|) and an [=origin=] (|origin|), this algorithm returns a
+  new [=Permissions Policy=].
 
-  1. Let |inherited policy| be a new [=ordered map=].
+  1. [=Assert=]: |container| is a [=fenced navigable container=].
 
-  1. Let |effective permissions| be an empty [=list=].
-  
   1. Let |fenced frame config| be |container|'s [=Node/node document=]'s [=navigable/active browsing
      context=]'s [=browsing context/fenced frame config instance=].
 
-  1. If |fenced frame config| is not null, and |fenced frame config|'s [=fenced frame config/
-     effective enabled permissions=] are not null, set |effective permissions| to
-     |fenced frame config|'s [=fenced frame config/effective enabled permissions=].
+  1. Let |inherited policy| be a new [=ordered map=].
+
+  1. If |fenced frame config| is not null and |fenced frame config|'s [=fenced frame config
+     instance/permissions policy behavior=] is [=permissions policy behavior/fixed=], then:
+
+    1. [=list/For each=] |feature| [=supported features|supported=]:
+
+        1. If |fenced frame config|'s [=fenced frame config instance/effective enabled permissions=]
+           [=list/contains=] |feature|, then set |inherited policy|[feature] to "`Enabled`".
+
+           Otherwise, set |inherited policy|[feature] to "`Disabled`".
 
-  1. [=list/For each=] |feature| [=supported features|supported=]:
+        Note: While this doesn't take the <{fencedframe/allow}> attribute into consideration, it
+        will have already been checked by the time this is called because of [=Should navigation
+        response to navigation request be blocked by Permissions Policy?=]. Any policy specified in
+        <{fencedframe/allow}> that is too restrictive would have cause the fenced frame to not load,
+        and any policy that is more permissive than what is specified in the [=fenced frame
+        config/effective enabled permissions=] will be ignored.
 
-    1. If |effective permissions| [=list/contains=] |feature|, then set |inherited policy|[feature]
-       to "`Enabled`".
-       
-       Otherwise, set |inherited policy|[feature] to "`Disabled`".
+  1. Otherwise:
+
+    1. [=list/For each=] |feature| [=supported features|supported=]:
 
+      1. If |feature| matches one of the [=fenced frame allowed permissions=], then set
+        |inherited policy|[feature] to the result of running [$Define an inherited policy for
+        feature in container at origin$] given |feature|, |container|, and |origin|.
+
+         Otherwise, set |inherited policy|[feature] to "`Disabled`".
+    
   1. Let |policy| be a new [=permissions policy=], with [=permissions policy/inherited policy=]
-     |inherited policy| and [=permissions policy/declared policy=] a new [=ordered map=].
+     |inherited policy| and [=permissions policy/declared policy=] «[], []».
```

Comment:
I don't think this change is valid. I'd say "initialized to two new [=ordered map=]s", since a declared policy is a struct with two ordered maps, but you've specified something that isn't quite valid. A single empty map would be `«[]»`, but you're using `«»` as if it were struct-literal syntax which doesn't exist. I think you mean something like `{«[]», «[]»}` (i.e., an "object"/struct that contains two empty maps), except `{}` syntax isn't real, so I'd just explicitly mention "initialized to two new ordered maps".

Reply:
I've replaced that with the phrase "initialized to two new ordered maps". If that's still too vague, I can explicitly create a new |declared policy| variable in a separate step and this to that variable's value.